File: Samples/VB.Net/VBDotNet/IDatabaseDlg.vb
Imports Microsoft.VisualBasic
Imports System
Imports System.Collections.Generic
Imports System.ComponentModel
Imports System.Data
Imports System.Drawing
Imports System.Text
Imports System.Windows.Forms
Imports SalesMatePlusLib
Imports System.Data.SqlClient

Namespace CSharpSDKTest

	Partial Public Class IDatabaseDlg
		Inherits Form

		Public Sub New()
			InitializeComponent()
		End Sub


		Private Sub Adbtn_Click(ByVal sender As Object, ByVal e As EventArgs) Handles Adbtn.Click
			If FldNmeTxt.Text = "" Then
				MessageBox.Show("Enter the Field Name")
			Else
				FLdNmeListBox.Items.Add(FldNmeTxt.Text)

				FldNmeTxt.Clear()
			End If
		End Sub

		Private Sub Clsbtn_Click(ByVal sender As Object, ByVal e As EventArgs) Handles Clsbtn.Click
			Me.Close()
		End Sub
		Private Sub Exebtn_Click(ByVal sender As Object, ByVal e As EventArgs) Handles Exebtn.Click
			Try
				Dim m_Database As IDatabase
				m_Database = New DatabaseClass()

				Dim sCommand As String = String.Empty

				If MSaccesbtn.Checked = True Then
					Dim iMsAcess As Integer
					iMsAcess= m_Database.IsMSAccess
				End If

				If opertncmbobox.Text = "SELECT" Then
					Dim objconnection As Object = m_Database.ConnectionInterface
					Dim strDataBase As String = Tabletxt.Text
					Dim ds As New DataSet()

					For icount As Integer = 0 To FLdNmeListBox.Items.Count - 1
						sCommand = "SELECT "
						Dim strFldName As String = FLdNmeListBox.Items(icount).ToString()

						sCommand &= " " & strFldName & " "

						sCommand &= " FROM " & strDataBase
						Dim outRecordset As Object = New Object()
						m_Database.ExecuteQueryEx(sCommand, outRecordset)
						Dim objRec As ADODB.Recordset = CType(outRecordset, ADODB.Recordset)
						Dim da As New System.Data.OleDb.OleDbDataAdapter()
						da.Fill(ds, objRec, strFldName)
						RsltGrid.DataSource = ds.DefaultViewManager
						sCommand = String.Empty

					Next icount
				End If
			Catch Ex As Exception
				MessageBox.Show(Ex.ToString())
			End Try

		End Sub

		Private Sub IDatabaseDlg_Load(ByVal sender As Object, ByVal e As EventArgs) Handles MyBase.Load

			Dim m_Database As IDatabase
			m_Database = New DatabaseClass()
			DBPathlist.Items.Add(m_Database.DBPath)
		End Sub

		Private Sub MSaccesbtn_CheckedChanged(ByVal sender As Object, ByVal e As EventArgs) Handles MSaccesbtn.CheckedChanged
			If MSaccesbtn.Checked = True Then
				Sqlbtn.Checked = False
			End If
		End Sub

		Private Sub Sqlbtn_CheckedChanged(ByVal sender As Object, ByVal e As EventArgs) Handles Sqlbtn.CheckedChanged
			If Sqlbtn.Checked=True Then
				MSaccesbtn.Checked=False
			End If

		End Sub
	End Class

End Namespace

File: Samples/C#.Net/CSharpSDKTest/IDatabaseDlg.cs
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Data;
using System.Drawing;
using System.Text;
using System.Windows.Forms;
using SalesMatePlusLib;
using System.Data.SqlClient;

namespace CSharpSDKTest
{

    public partial class IDatabaseDlg : Form
    {

        public IDatabaseDlg()
        {
            InitializeComponent();
        }


        private void Adbtn_Click(object sender, EventArgs e)
        {
            if (FldNmeTxt.Text == "")
            {
                MessageBox.Show("Enter the Field Name");
            }
            else
            {
                FLdNmeListBox.Items.Add(FldNmeTxt.Text);

                FldNmeTxt.Clear();
            }
        }

        private void Clsbtn_Click(object sender, EventArgs e)
        {
            this.Close();
        }
        private void Exebtn_Click(object sender, EventArgs e)
        {
            try
            {
                IDatabase m_Database;
                m_Database = new DatabaseClass();
                
                string sCommand = string.Empty;

                if (MSaccesbtn.Checked == true)
                {
                    int iMsAcess;
                    iMsAcess= m_Database.IsMSAccess;
                }
               
                if (opertncmbobox.Text == "SELECT")
                {
                    object objconnection = m_Database.ConnectionInterface;
                    string strDataBase = Tabletxt.Text;
                    DataSet ds = new DataSet();

                    for (int icount = 0; icount < FLdNmeListBox.Items.Count; icount++)
                    {
                        sCommand = "SELECT ";
                        string strFldName = FLdNmeListBox.Items[icount].ToString();

                        sCommand += " " + strFldName + " ";

                        sCommand += " FROM " + strDataBase;
                        object outRecordset = new object();
                        m_Database.ExecuteQueryEx(sCommand, ref outRecordset);
                        ADODB.Recordset objRec = (ADODB.Recordset)outRecordset;
                        System.Data.OleDb.OleDbDataAdapter da =new System.Data.OleDb.OleDbDataAdapter();
                        da.Fill(ds, objRec, strFldName);
                        RsltGrid.DataSource = ds.DefaultViewManager;
                        sCommand = string.Empty;

                     }
                }
            }
            catch (Exception Ex)
            {
                MessageBox.Show(Ex.ToString());
            }
           
        }

        private void IDatabaseDlg_Load(object sender, EventArgs e)
        {

            IDatabase m_Database;
            m_Database = new DatabaseClass();
            DBPathlist.Items.Add(m_Database.DBPath);
        }

        private void MSaccesbtn_CheckedChanged(object sender, EventArgs e)
        {
            if (MSaccesbtn.Checked == true)
            {
                Sqlbtn.Checked = false;
            }
        }

        private void Sqlbtn_CheckedChanged(object sender, EventArgs e)
        {
            if(Sqlbtn.Checked==true)
            {
                MSaccesbtn.Checked=false;
            }

        }
    }

}